File: server/src/main/java/com/denfense/server/dto/breeding/MythicBreedingDtos.java
package com.denfense.server.dto.breeding;

import java.time.Instant;
import java.util.List;

public final class MythicBreedingDtos {
    private MythicBreedingDtos() {}
    public record Slot(int slotNo, String status, String unlockSource, Instant startedAt, Instant readyAt) {}
    public record Candidate(Long userAlienId, Long alienId, String name, int level, boolean selectable) {}
    public record UnlockRequest(String requestId) {}
    public record StartRequest(Long parentUserAlienIdA, Long parentUserAlienIdB, String requestId) {}
    public record ClaimRequest(String requestId) {}
    public record AccelerateRequest(String requestId, int units) {}
    public record SlotsResponse(List<Slot> slots, int accountLevel, int diamond, int slot2UnlockLevel,
                                int slot2GemPrice, int slot3GemPrice, int durationSeconds,
                                int accelerationUnitSeconds, int accelerationUnitDiamondCost) {}
    public record CandidatesResponse(List<Candidate> candidates) {}
    public record StartResponse(int slotNo, String status, Instant readyAt) {}
    public record ClaimResponse(int slotNo, long resultAlienId, String status, Instant claimedAt) {}
    public record AccelerateResponse(int slotNo, String status, int requestedUnits, int appliedUnits,
                                     int spentDiamond, int remainingDiamond, Instant readyAt) {}
}

File: Client/Assets/Scripts/Ui/Alien/MythicBreedingDtos.cs
using System;

[Serializable] public sealed class MythicBreedingSlotDto { public int slotNo; public string status; public string unlockSource; public string startedAt; public string readyAt; }
[Serializable] public sealed class MythicBreedingSlotsResponse { public MythicBreedingSlotDto[] slots; public int accountLevel; public int diamond; public int slot2UnlockLevel; public int slot2GemPrice; public int slot3GemPrice; public int durationSeconds; public int accelerationUnitSeconds; public int accelerationUnitDiamondCost; }
[Serializable] public sealed class MythicBreedingCandidateDto { public long userAlienId; public long alienId; public string name; public int level; public bool selectable; }
[Serializable] public sealed class MythicBreedingCandidatesResponse { public MythicBreedingCandidateDto[] candidates; }
[Serializable] public sealed class MythicBreedingUnlockRequest { public string requestId; }
[Serializable] public sealed class MythicBreedingStartRequest { public long parentUserAlienIdA; public long parentUserAlienIdB; public string requestId; }
[Serializable] public sealed class MythicBreedingClaimRequest { public string requestId; }
[Serializable] public sealed class MythicBreedingAccelerateRequest { public string requestId; public int units; }
[Serializable] public sealed class MythicBreedingStartResponse { public int slotNo; public string status; public string readyAt; }
[Serializable] public sealed class MythicBreedingClaimResponse { public int slotNo; public long resultAlienId; public string status; public string claimedAt; }
[Serializable] public sealed class MythicBreedingAccelerateResponse { public int slotNo; public string status; public int requestedUnits; public int appliedUnits; public int spentDiamond; public int remainingDiamond; public string readyAt; }
[Serializable] public sealed class MythicBreedingRecipeDocument { public MythicBreedingRecipeDto[] recipes; }
[Serializable] public sealed class MythicBreedingRecipeDto
{
    public string recipeKey;
    public long parentAlienIdA;
    public long parentAlienIdB;
    public long[] standardResultAlienIds;
    public int standardWeightEach;
    public long exclusive19AlienId;
    public int exclusive19Weight;
    public long exclusive20AlienId;
    public int exclusive20Weight;
    public bool enabled;
}
